test/unit/connector-test.js

```diff
@@ -452,3 +454,42 @@ exports['ParallelConnectionStrategy'] = {
     });
   }
 };
+
+exports['Test unicode SQL Server name'] = {
+  setUp: function (done) {
+    this.sinon = sinon.sandbox.create();
```

Comment:
I don't think using Sinon :bowtie: here is necessary, as we are checking if dns.lookup is called with the server name, test could be simpler :sweat_smile: something like:

```js
  'testIDN': function(test) {
    const server = '本地主机.ad';
    const connector = new Connector({ host: server, port: 12345 }, false);
    connector.executeForHostname((err) => { test.ifError(err);});
    test.done();
  }
```